connectors/http/polling/src/main/java/io/camunda/connector/http/polling/task/PollingTask.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.http.polling.task;
+
+import io.camunda.connector.api.inbound.Health;
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.http.base.model.HttpCommonRequest;
+import io.camunda.connector.http.base.model.HttpCommonResult;
+import io.camunda.connector.http.base.services.HttpService;
+import io.camunda.connector.http.polling.DeactivationHandler;
+import java.io.IOException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PollingTask implements Runnable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PollingTask.class);
+
+  private final InboundIntermediateConnectorContext context;
+  private final HttpService httpService;
+  private final DeactivationHandler deactivationHandler;
+
+  public PollingTask(
+      InboundIntermediateConnectorContext context,
+      HttpService httpService,
+      DeactivationHandler deactivationHandler) {
+    this.context = context;
+    this.httpService = httpService;
+    this.deactivationHandler = deactivationHandler;
+  }
+
+  @Override
+  public void run() {
+    try {
+      if (isReadyForBinding()) {
+        Optional.ofNullable(bindContextToRequest())
+            .map(this::executeHttpRequest)
+            .ifPresent(this::correlateHttpResponse);
+      }
```

Comment:
We only should have a poll task for instances that are ready as discussed before.